src/Workspaces/CoreTest/Host/WorkspaceServices/TestTemporaryStorageServiceFactory.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.UnitTests.Persistence
+{
+    [ExportWorkspaceServiceFactory(typeof(ITemporaryStorageService), ServiceLayer.Test), Shared, PartNotDiscoverable]
+    internal sealed class TestTemporaryStorageServiceFactory : IWorkspaceServiceFactory
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public TestTemporaryStorageServiceFactory()
+        {
+        }
+
+        public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
+            => TrivialTemporaryStorageService.Instance;
```

Comment:
is there a non-trivial storage service? :)  can/should we test that in tests instead?

Reply:
TrivialTemporaryStorageService is one of our two production-supported services.

Reply:
The test factory ensures a consistent service is used in tests regardless of platform:

https://github.com/dotnet/roslyn/blob/b2d1f8152f7cbb7a88ad9907cc0271498f9d8c0d/src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs#L36-L41

Reply:
Seems like we should be using that? Otherwise we're not testing what we ship?